v1/topdown/cache/cache.go

```diff
@@ -24,10 +24,23 @@ const (
 	defaultStaleEntryEvictionPeriodSeconds   = int64(0)   // never
 )
 
+var interQueryBuiltinValueCacheDefaultConfigs = map[string]*InterQueryBuiltinValueCacheConfig{}
+
+func getDefaultInterQueryBuiltinValueCacheConfig(name string) *InterQueryBuiltinValueCacheConfig {
+	return interQueryBuiltinValueCacheDefaultConfigs[name]
+}
+
+// RegisterDefaultInterQueryBuiltinValueCacheConfig registers a default configuration for the inter-query value cache;
+// used when none has been explicitly configured.
+// To disable a named cache when not configured, pass a nil config.
+func RegisterDefaultInterQueryBuiltinValueCacheConfig(name string, config *InterQueryBuiltinValueCacheConfig) {
+	interQueryBuiltinValueCacheDefaultConfigs[name] = config
+}
+
 // Config represents the configuration for the inter-query builtin cache.
 type Config struct {
-	InterQueryBuiltinCache      InterQueryBuiltinCacheConfig      `json:"inter_query_builtin_cache"`
-	InterQueryBuiltinValueCache InterQueryBuiltinValueCacheConfig `json:"inter_query_builtin_value_cache"`
+	InterQueryBuiltinCache      InterQueryBuiltinCacheConfig          `json:"inter_query_builtin_cache"`
+	InterQueryBuiltinValueCache RootInterQueryBuiltinValueCacheConfig `json:"inter_query_builtin_value_cache"`
```

Comment:
Isn't this a breaking change?

Reply:
Hm, yes. I suppose we could break the global config and the named configs apart to make this non-breaking 🤔 .